history/storage.py

```diff
@@ -187,41 +209,43 @@ def store_response(self, spider, request, response):
         s3_key = self.s3_bucket.new_key(key)
 
         try:
-            #s3_key.update_metadata(metadata) #=> can't use this as need to cast to unicode
+            # s3_key.update_metadata(metadata) #=> can't use this as need to cast to unicode
             for k, v in metadata.items():
                 if isinstance(v, str):
                     v = v[:400] + '...' if len(v) > 400 else v
                 s3_key.set_metadata(k, unicode(v))
             s3_key.set_contents_from_string(data_string)
-
-            #save source file
+            # save source file
             if self.SAVE_SOURCE:
```

Comment:
`Else` ? Shouldn't we yield a `NotConfigured` at the beginning ?

This function could gain a lot in clarity by hiding some of the steps behinds methods.